contracts/bancor-portal/BancorPortal.sol

```diff
@@ -105,33 +108,36 @@ contract BancorPortal is IBancorPortal, ReentrancyGuardUpgradeable, Utils, Upgra
         IBancorNetwork network,
         INetworkSettings networkSettings,
         IERC20 bnt,
-        IUniswapV2Router02 uniswapV2Router,
-        IUniswapV2Factory uniswapV2Factory,
-        IUniswapV2Router02 sushiSwapV2Router,
-        IUniswapV2Factory sushiSwapV2Factory
+        IUniswapV2Router02 initUniswapV2Router,
+        IUniswapV2Factory initUniswapV2Factory,
+        IUniswapV2Router02 initSushiSwapV2Router,
+        IUniswapV2Factory initSushiSwapV2Factory,
+        IERC20 initWETH
     )
         validAddress(address(network))
         validAddress(address(networkSettings))
         validAddress(address(bnt))
-        validAddress(address(uniswapV2Router))
-        validAddress(address(uniswapV2Factory))
-        validAddress(address(sushiSwapV2Router))
-        validAddress(address(sushiSwapV2Factory))
+        validAddress(address(initUniswapV2Router))
+        validAddress(address(initUniswapV2Factory))
+        validAddress(address(initSushiSwapV2Router))
+        validAddress(address(initSushiSwapV2Factory))
+        validAddress(address(initWETH))
     {
         _network = network;
         _networkSettings = networkSettings;
         _bnt = bnt;
-        _uniswapV2Router = uniswapV2Router;
-        _uniswapV2Factory = uniswapV2Factory;
-        _sushiSwapV2Router = sushiSwapV2Router;
-        _sushiSwapV2Factory = sushiSwapV2Factory;
+        _uniswapV2Router = initUniswapV2Router;
+        _uniswapV2Factory = initUniswapV2Factory;
+        _sushiSwapV2Router = initSushiSwapV2Router;
+        _sushiSwapV2Factory = initSushiSwapV2Factory;
+        _weth = initWETH;
```

Comment:
You can also get it by calling `initUniswapV2Router.WETH()`. Might be slightly cleaner, since it won't require us to add another explicit argument (which is more error-prone since it can be accidentally 0x0 or different than `initUniswapV2Router.WETH()`.

Alternatively (or in addition), we can verify that `initWETH == initUniswapV2Router.WETH() == initSushiSwapV2Router.WETH()`. It is correct on mainnet but may require more tweaking on testnets or other chains, so it's better safe than sorry.

Reply:
done